src/languages/sqf.js

```diff
@@ -2473,6 +2503,10 @@ export default function(hljs) {
       STRINGS,
       PREPROCESSOR
     ],
-    illegal: /\w\$|\?|\@/
+    illegal: [
+      '\w\$',  //$ is only valid when used with Hex numbers (e.g. $FF) or at the beginning of localized strings (e.g. "$STR_HELLO")
+      '\?',    //There's no ? in SQF
+      '\@/'    //There's no @ in SQF
```

Comment:
These should be regex:

```suggestion
      /\w\$/,  //$ is only valid when used with Hex numbers (e.g. $FF) or at the beginning of localized strings (e.g. "$STR_HELLO")
      /\?/,    //There's no ? in SQF
      /@\//    //There's no @ in SQF
```

Why does @ need to be escaped?

Reply:
I don't know. It doesn't?

Reply:
@joshgoebel 
```js
illegal: [
      /\w\$/,  //$ is only valid when used with Hex numbers (e.g. $FF) or at the beginning of localized strings (e.g. "$STR_HELLO")
      '?'   ,  //There's no ? in SQF
      '@'      //There's no @ in SQF
    ]
```
is this better?

Reply:
No, they should be regex (not strings).  The `?` does need to be escaped (as I showed)I since `?` is a quantifier.  Strings are just "raw regex" (escaping is not performed) and we prefer actual regex whenever possible to avoid this type of confusion.

Reply:
Does `@` mean something special in regex that I'm just not thinking of?

Reply:
I don't know, but both escaped and non-escaped versions match correctly. So I'm assuming it's not needed as you said.